lib/node-storage.ts

```diff
@@ -0,0 +1,78 @@
+/*
+Copyright 2020 Balena Ltd.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import { promises as fs } from 'fs';
+import * as path from 'path';
+import { StorageLike } from './local-storage';
+
+export class NodeStorage implements StorageLike {
+	private initialized = false;
+	constructor(private dataDirectory: string) {}
+
+	public async init() {
+		if (this.initialized) {
+			return;
+		}
+		this.initialized = true;
```

Comment:
Doesn't this need to be `boolean | Promise` (holding the `mkdir()` promise)?
I think that atm there can be a race condition for two concurrent `setItem` calls where the second one calls `init` and then proceeds attempting to store the result, before the `mkdir()` of the `init()` of the first one finishes.